src/main/java/io/supertokens/emailpassword/PasswordHashingUtils.java

```diff
@@ -17,12 +17,43 @@
 package io.supertokens.emailpassword;
 
 import io.supertokens.config.CoreConfig;
+import com.lambdaworks.crypto.SCrypt;
 import io.supertokens.emailpassword.exceptions.UnsupportedPasswordHashingFormatException;
+import org.apache.tomcat.util.codec.binary.Base64;
 
 import javax.annotation.Nullable;
+import javax.crypto.Cipher;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import java.nio.charset.StandardCharsets;
+import java.security.*;
+import java.util.Objects;
 
 public class PasswordHashingUtils {
 
+    public static class ParsedFireBaseSCryptResponse {
+        String passwordHash;
+        String salt;
+        String saltSeparator;
+        int rounds;
+        int memCost;
+
+        public ParsedFireBaseSCryptResponse(String passwordHash, String salt, String saltSeparator, int rounds,
+                int memCost) {
+            this.passwordHash = passwordHash;
+            this.salt = salt;
+            this.saltSeparator = saltSeparator;
+            this.rounds = rounds;
+            this.memCost = memCost;
+        }
+    }
+
+    private static final String FIREBASE_SCRYPT_PREFIX = "f_scrypt";
+    private static final String FIREBASE_SCRYPT_SEPARATOR = "\\$";
```

Comment:
why the `\\`?

Reply:
`split` takes a regex, so `$` needs to be escaped